converters/fromarray/instrumentation.ts

```diff
@@ -0,0 +1,90 @@
+let instrumentationConverters = [];
+instrumentationConverters[Instrumentation.JsError] = jsErrorFromArray;
+instrumentationConverters[Instrumentation.MissingFeature] = missingFeatureFromArray;
+instrumentationConverters[Instrumentation.XhrError] = xhrErrorFromArray;
+instrumentationConverters[Instrumentation.TotalByteLimitExceeded] = byteLimitExceededFromArray;
+instrumentationConverters[Instrumentation.Teardown] = noDataFromArray;
+instrumentationConverters[Instrumentation.PerformanceStateError] = noDataFromArray;
+instrumentationConverters[Instrumentation.ClarityAssertFailed] = assertFailedFromArray;
+instrumentationConverters[Instrumentation.ClarityDuplicated] = clarityDuplicatedFromArray;
+instrumentationConverters[Instrumentation.ClarityActivateError] = clarityActivateErrorFromArray;
+instrumentationConverters[Instrumentation.ShadowDomInconsistent] = inconsistentShadowDomFromArray;
+
+export default instrumentationConverters;
+
+function jsErrorFromArray(jsErrorData: any[]): IJsErrorEventData {
+  let data: IJsErrorEventData = {
+    source  : jsErrorData[0],
+    message : jsErrorData[1],
+    stack   : jsErrorData[2],
+    lineno  : jsErrorData[3],
+    colno   : jsErrorData[4]
```

Comment:
Could you not encode this mapping in a data structure and make the code generic? Or code-gen it if you are concerned about performance.